### PR TITLE
Add io.github.roseblume.rosemusic

### PR DESCRIPTION
**Pull request by RoseBlume**
### Please confirm your submission meets all the criteria

- [X] Please describe the application briefly.
  
A simple online radio player with a mobile friendly UI
- [x] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read the and followed all the [Submission and licence requirements][reqs].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

**Comment by hfiguiere** [comment]
>  Please describe the application briefly.

this is missing.

**Comment by RoseBlume** [comment]
Both have been fixed.

File: io.github.roseblume.rosemusic.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  - --socket=pulseaudio
+modules:
+  - name: rose-music
+    buildsystem: simple
+    build-commands:
+      - ar -x *.deb
+      - tar -xf data.tar.gz
+      - install -Dm755 usr/bin/rosary-music /app/bin/rosary-music
+      - sed -i -e 's|Categories=|Categories=Education|g' usr/share/applications/Rosary-Music.desktop
+      - sed -i -e 's|Icon=rosary-music|Icon=io.github.roseblume.rosemusic|g' usr/share/applications/Rosary-Music.desktop
```
Comment (hfiguiere) [inline io.github.roseblume.rosemusic.yaml:21]: don't use sed on the desktop file. There is the `desktop-file-edit` command.

File: io.github.roseblume.rosemusic.metainfo.xml
Comment (hfiguiere) [inline io.github.roseblume.rosemusic.metainfo.xml:1]: this file should be upstream.

File: io.github.roseblume.rosemusic.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      - install -Dm644 usr/share/icons/hicolor/128x128/apps/rosary-music.png /app/share/icons/hicolor/128x128/apps/io.github.roseblume.rosemusic.png
+      - install -Dm644 usr/share/icons/hicolor/32x32/apps/rosary-music.png /app/share/icons/hicolor/32x32/apps/io.github.roseblume.rosemusic.png
+      - install -Dm644 usr/share/icons/hicolor/256x256@2/apps/rosary-music.png /app/share/icons/hicolor/256x256@2/apps/io.github.roseblume.rosemusic.png
+      - install -Dm644 io.github.roseblume.rosemusic.metainfo.xml /app/share/metainfo/io.github.roseblume.rosemusic.metainfo.xml
+    sources:
+      - type: file
+        url: https://github.com/RoseBlume/Rose-Music/releases/download/v1.0.0/Rosary-Music_1.0.0_amd64.deb
+        sha256: f3b175139e6947f59db4f2c342c21fba2106df7a6056ce6bf5a05fa8634f8d56
+        dest-filename: rosemusic.deb
+        only-arches: [x86_64]
```
Comment (hfiguiere) [inline io.github.roseblume.rosemusic.yaml:32]: so what happen in the aarch64 build? why not building from source to produce an aarch64 build?

Comment (RoseBlume) [reply io.github.roseblume.rosemusic.yaml:31]: Unfortunately building from source is not possible without significantly increasing the applications size.

Comment (hfiguiere) [reply io.github.roseblume.rosemusic.yaml:31]: I don't why the application size would change. But definitely building from source is preferred.

Comment (RoseBlume) [reply io.github.roseblume.rosemusic.yaml:31]: I would have to switch to freedesktop and provide webkit2gtk

File: io.github.roseblume.rosemusic.metainfo.xml
Comment (RoseBlume) [reply io.github.roseblume.rosemusic.metainfo.xml:1]: Its now upstreamed.

**Comment by RoseBlume** [comment]
This should be good now.

**Review by hfiguiere — CHANGES_REQUESTED** [review]
This was left unsolved.
https://github.com/flathub/flathub/pull/5761#discussion_r1823087068

File: io.github.roseblume.rosemusic.yaml (hunk 2 above)
Comment (hfiguiere) [reply io.github.roseblume.rosemusic.yaml]: I don't see the reason. Maybe you just don't need the GNOME in the end, which is a super set of the freedesktop SDK.

I reiterate. You should build from source.

Comment (RoseBlume) [reply io.github.roseblume.rosemusic.yaml]: With Freedesktop  I would have to provide webkit2gtk. I have already seen this in practice and it results in a very large bundle.

Comment (hfiguiere) [reply io.github.roseblume.rosemusic.yaml]: why would you have to use the freedesktop bundle? You still don't explain, just handwave.

Comment (RoseBlume) [reply io.github.roseblume.rosemusic.yaml]: I would need both Rust and NodeJS in order to compile. In order to get them I would need to use the freedesktop sdk.

Comment (hfiguiere) [reply io.github.roseblume.rosemusic.yaml]: That's not true at all.

Comment (RoseBlume) [reply io.github.roseblume.rosemusic.yaml]: Where else can I get them?

Comment (hfiguiere) [reply io.github.roseblume.rosemusic.yaml]: they are the exact same. There are many apps using the GNOME Sdk that use either or a combination of these SDK extensions. There is no difference. Just make sure they are installed.

Comment (RoseBlume) [reply io.github.roseblume.rosemusic.yaml]: Alright thanks, I'll give it a shot.

**Comment by RoseBlume** [comment]
bot, build io.github.roseblume.rosemusic

**Comment by RoseBlume** [comment]
bot, build io.github.roseblume.rosemusic

**Comment by RoseBlume** [comment]
Can someone please take a look at my main repo and help me out? I got past the npm part but install the dependencies with Cargo is still failing.
Repo: https://github.com/RoseBlume/RoseMusic
Workflow Run: https://github.com/RoseBlume/RoseMusic/actions/runs/11605870712

**Comment by RoseBlume** [comment]
bot, build io.github.roseblume.rosemusic

**Comment by RoseBlume** [comment]
Weird. Failed to  create upload token.

**Comment by RoseBlume** [comment]
bot, build io.github.roseblume.rosemusic

**Comment by RoseBlume** [comment]
@hfiguiere What does this mean?

**Comment by hfiguiere** [comment]
it mean that you have to be patient until this gets fixed. Nothing I can do but those that can have been notified.

**Comment by RoseBlume** [comment]
bot, build io.github.roseblume.rosemusic

**Comment by RoseBlume** [comment]
bot, build io.github.roseblume.rosemusic

**Comment by RoseBlume** [comment]
bot, build io.github.roseblume.rosemusic

**Comment by RoseBlume** [comment]
bot, build io.github.roseblume.rosemusic

**Comment by RoseBlume** [comment]
bot, build io.github.roseblume.rosemusic

**Comment by RoseBlume** [comment]
bot, build io.github.roseblume.rosemusic

File: io.github.roseblume.rosemusic.yaml (hunk 2 above)
Comment (RoseBlume) [reply io.github.roseblume.rosemusic.yaml]: I got it to compile now I just have issues with committing the build.

Comment (hfiguiere) [reply io.github.roseblume.rosemusic.yaml]: there are some infra issue in that regard. please be patient. We can retrigger a build later when issues are resolved.

**Comment by RoseBlume** [comment]
@hfiguiere Im sorry to bother you but the builds work fine but it fails in committing them. What can cause this, and what are some possible solutions?

**Comment by hfiguiere** [comment]
https://status.flathub.org/

**Comment by RoseBlume** [comment]
Thank you.

**Comment by bbhtt** [comment]
bot, build io.github.roseblume.rosemusic

**Pull request closed without merging**